src/common/tusb_compiler.h

```diff
@@ -175,6 +175,11 @@
 		#pragma GCC poison tud_vendor_control_request_cb
 	#endif
 
+  #if defined (__clang__)
```

Comment:
can you do the #ifdef #else instead of undef (ust move this up). undef is rather confusing since people may only look at the first define.

Reply:
Just to clarify @hathach 

Like this?
```
#if defined(__GNUC__) || defined (__clang__)
  #define TU_ATTR_ALIGNED(Bytes)        __attribute__ ((aligned(Bytes)))
  #define TU_ATTR_SECTION(sec_name)     __attribute__ ((section(#sec_name)))
  #define TU_ATTR_PACKED                __attribute__ ((packed))
  #if defined(__GNUC__)
    #define TU_ATTR_WEAK                  __attribute__ ((weak))
  #else
    #define TU_ATTR_WEAK                  __attribute__ ((weak_import))
  #endif
... 
...
```

Reply:
yeah, but the otherway around

```C
  #if defined(__clang__)
    #define TU_ATTR_WEAK                  __attribute__ ((weak_import))
  #else
    #define TU_ATTR_WEAK                  __attribute__ ((weak))
  #endif
```

Reply:
All fixed :)

Reply:
unit test probably compile using clang, and has affected by this attribute. I will check later.